codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieHeaderNames.java

```diff
@@ -30,6 +30,37 @@
 
     public static final String SAMESITE = "SameSite";
 
+    /**
+     * Possible values for the SameSite attribute.
+     * See <a href=https://tools.ietf.org/html/draft-west-cookie-incrementalism-00#section-3.1>changes to RFC6265bis</a>
+     */
+    public enum SameSite {
+        Lax,
+        Strict,
+        None;
+
+        /**
+         * Return the enum value corresponding to the passed in same-site-flag, using a case insensitive comparison.
+         *
+         * @param name value for the SameSite Attribute
+         * @return enum value for the provided name or null
+         */
+        static SameSite of(String name) {
+            if (name != null) {
+                try {
+                    return SameSite.valueOf(name);
+                } catch (IllegalArgumentException e) {
+                    for (SameSite each : SameSite.class.getEnumConstants()) {
+                        if (each.name().equalsIgnoreCase(name)) {
+                            return each;
+                        }
+                    }
```

Comment:
should we regrow the `e` if we did not find a match ?

Reply:
I've implemented it as per Bryce's suggestion that we ignore illegal values. I would think it makes sense as it seems you are not throwing exception for any validation errors in the Decoder class even when strict validation is enabled (cookie is just ignored).

However I might have missed anything and, in any case, I will follow your advice. My initial idea was to change "logger" in the CookieDecoder class to be protected and add logging but let it return null anyway.

Reply:
good point... I think logging would be good tho :)

Reply:
might be nicer to add a name field to the enum itself so you can use upper case for the enum names..

```java
public enum SameSite {
        LAX("Lax"),
        STRICT("Strict"),
        NONE("None");

        private final String name;
```